### PR TITLE
Fix docstrings in VQR

### LINKED ISSUE #577: VQR `estimator` parameter indicates incorrect default
>estimator (BaseEstimator | None) – An estimator to be used to evaluate expectation values of the observable. If None the [qiskit.primitives.BaseEstimator](https://qiskit.org/documentation/stubs/qiskit.primitives.BaseEstimator.html#qiskit.primitives.BaseEstimator) is used. The underlying QNN is [EstimatorQNN](https://qiskit.org/documentation/machine-learning/stubs/qiskit_machine_learning.neural_networks.EstimatorQNN.html#qiskit_machine_learning.neural_networks.EstimatorQNN).

This should be the `Estimator` class there not the abstract parent base class `BaseEstimator` type

### PR DESCRIPTION
**Pull request by adekusar-drl**
### Summary
Fixes #577 .

### Details and comments
Now the docstring states:
```
estimator: An estimator to be used to evaluate expectation values of the observable.
    If ``None`` the :class:`qiskit.primitives.Estimator` is used. If an estimator
    instance is set, the underlying QNN will be of type
    :class:`~qiskit_machine_learning.neural_networks.EstimatorQNN`, and the estimator
    primitive will be used to compute the neural network's results.
```

### CODE REVIEW

File: qiskit_machine_learning/algorithms/regressors/vqr.py [hunk 1]
```diff
@@ -88,9 +88,8 @@ def __init__(
                 as an array and a computed value as a float of the objective function being
                 optimized. This allows to track how well optimization / training process is going on.
             estimator: An estimator to be used to evaluate expectation values of the observable.
-                If ``None`` the :class:`qiskit.primitives.BaseEstimator` is used. The underlying QNN
-                is :class:`~qiskit_machine_learning.neural_networks.EstimatorQNN`.
-            If an estimator instance is set, the underlying QNN will be of type
+                If ``None`` the :class:`qiskit.primitives.Estimator` is used. If an estimator
+                instance is set, the underlying QNN will be of type
```
Comment (woodsp-ibm) [inline qiskit_machine_learning/algorithms/regressors/vqr.py:92]: I noticed the BaseEstimator when answering a question, but in reading this again I note that we say
> If an estimator instance is set, the underlying QNN will be of type....

but arguably the underlying type is always EstimatorQNN, whether an `estimator is instance is set` or not - the default estimator stated here is that of that class (unlikely to change I guess). Maybe we can drop the bit about an instance being set?

Comment (adekusar-drl) [reply qiskit_machine_learning/algorithms/regressors/vqr.py]: What about this?
```
            estimator: An estimator to be used to evaluate expectation values of the observable.
                If ``None`` and quantum instance is not set, the 
                :class:`qiskit.primitives.Estimator` is used, the underlying QNN will be of type
                :class:`~qiskit_machine_learning.neural_networks.EstimatorQNN`, and the estimator
                primitive will be used to compute the neural network's results.
```

Comment (woodsp-ibm) [reply qiskit_machine_learning/algorithms/regressors/vqr.py]: I might do a full stop not a comma, after `is used` thus

> If ``None`` and quantum instance is not set, the  :class:`qiskit.primitives.Estimator` is used. 
> The underlying QNN will be of type....

so as it does not seem the latter part is simply qualifying that initial text so it still only applies in the None case.

Comment (adekusar-drl) [reply qiskit_machine_learning/algorithms/regressors/vqr.py]: Updated.

**Review by woodsp-ibm — APPROVED** [review]

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**